### PR TITLE
[ci] Enable native compiler in `egde:flambda` build.

### PR DESCRIPTION
**Pull request by ejgallego**
OCaml 4.07.0 should have fixed the [memory eating bug](https://caml.inria.fr/mantis/view.php?id=7630)

Depends on #8000

### CODE REVIEW

**Comment by SkySkimmer** [comment]
I assume the dependency is #7871?

**Comment by ejgallego** [comment]
> I assume the dependency is #7871?

Yup sorry.

**Comment by ejgallego** [comment]
Full travis build: https://travis-ci.org/ejgallego/coq/builds/402707917

**Comment by ppedrot** [comment]
This should probably also depend on #8000.

**Comment by ejgallego** [comment]
> This should probably also depend on #8000.

No need to; the stdlib compiles fine so far and the don't conflict.

**Comment by ejgallego** [comment]
> No need to; the stdlib compiles fine so far and the don't conflict.

What I ignore is if we ever test the result of the native compilation.

**Comment by ppedrot** [comment]
> What I ignore is if we ever test the result of the native compilation.

I doubt it, as the accumulator function which is used virtually everytime we have an open function does not work at all with flambda.

**Comment by ejgallego** [comment]
> I doubt it, as the accumulator function which is used virtually everytime we have an open function does not work at all with flambda.

So that seems like a critical bug on the test-suite, doesn't it?

**Comment by ppedrot** [comment]
@ejgallego Probably.

**Comment by ejgallego** [comment]
> So that seems like a critical bug on the test-suite, doesn't it?

I am wrong, indeed the test suite seems to fail.

File: dev/ci/docker/bionic_coq/Dockerfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ RUN opam init -a -y -j $NJOBS --compiler="$COMPILER" default https://opam.ocaml.
 # Common OPAM packages.
 # `num` does not have a version number as the right version to install varies
 # with the compiler version.
-ENV BASE_OPAM="num ocamlfind.1.8.0 jbuilder.1.0+beta20 ounit.2.0.8" \
+ENV BASE_OPAM="num ocamlfind.1.8.0 dune.1.0.0 ounit.2.0.8" \
     CI_OPAM="menhir.20180530 elpi.1.0.4 ocamlgraph.1.8.8"
```
Comment (MSoegtropIMC) [inline dev/ci/docker/bionic_coq/Dockerfile]: @ejgallego : can you explain the consequences of the jbuilder vs dune change? Is this related to #6857 ? Isn't in this PR intentionally?

Comment (ejgallego) [reply dev/ci/docker/bionic_coq/Dockerfile]: They are the same software, `jbuilder` got renamed to `dune` due to trademark issues.

**Review by SkySkimmer — APPROVED** [review]

**Comment by ejgallego** [comment]
Should be ready [provided CI works]

**Review by SkySkimmer — APPROVED** [review]
